apps/examples/wifi_manager_sample/wm_test.c

```diff
@@ -0,0 +1,625 @@
+	strcpy(ap_config.passphrase, ap_info->password);
+	ap_config.passphrase[strlen(ap_info->password)] = '\0';
+	ap_config.channel = 1;
+
+	printf("====================================\n");
+	printf("AP INFO\n");
+	printf("SSID: %s\n", ap_config.ssid);
+	printf("channel: %d\n", ap_config.channel);
+	printf("PASSWORD: %s\n", ap_config.passphrase);
+	printf("====================================\n");
+
+	res = wifi_manager_set_mode(SOFTAP_MODE, &ap_config);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		printf("[pkes] Run SoftAP Fail\n");
+	}
+	WM_TEST_LOG_END;
+}
+
+
+
+/****************************************************************************
+ * ocf_wifi_test
+ ****************************************************************************/
+
+void
+wm_auto_test(void *arg)
+{
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+
+	res = wifi_manager_init(&wifi_callbacks);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		printf("[pkes] wifi_manager_init fail\n");
+		return ;
+	}
+
+	wifi_manager_softap_config_s ap_config = { "PKES", 1, "1111122222" };
+	if (wifi_manager_set_softap_config(&ap_config) != WIFI_MANAGER_SUCCESS) {
+		printf("[pkes] cannot set ap config\n");
+		return ;
+	}
+
+	int cnt = 0;
+	while (cnt++ < WM_TEST_COUNT) {
+		printf("[pkes] T%d Starting round %d\n", getpid(), cnt);
+		res = wifi_manager_mode_set(SOFTAP_MODE);
+		if (res != WIFI_MANAGER_SUCCESS) {
+			printf("[pkes] Set AP mode Fail\n");
+			return ;
+		}
+		printf("[pkes] Step 1 done.. Waiting for STA join\n");
+
+		/* Wait station connect */
+		WM_TEST_WAIT;
+
+		printf("[pkes] Step 2 done..Waiting for STA leave\n");
+		/* Wait station disconnect */
+		WM_TEST_WAIT;
+
+		printf("[pkes] Step 3 done..Waiting for SoftAP down and startSTA\n");
+		// Do Soft AP down
+		res = wifi_manager_mode_set(STA_MODE);
+		if (res != WIFI_MANAGER_SUCCESS) {
+			printf("[pkes] Set AP mode Fail\n");
+			return ;
+		}
+		printf("[pkes] Step 4 done..Connecting to AP\n");
+		/* Connect to AP */
+		wifi_manager_ap_config_s apconfig = { "NETGEAR99", 9,
+											  "artikwifi", 9,
+											  WIFI_MANAGER_AUTH_WPA2_PSK,
+											  WIFI_MANAGER_CRYPTO_AES
+		};
+
+		res = wifi_manager_ap_connect(&apconfig);
+		if (res != WIFI_MANAGER_SUCCESS) {
+			printf("[pkes] Ap connect failed in round %d\n", cnt);
+			return ;
+		}
+		/* Wait for DHCP connection */
+		WM_TEST_WAIT;
+		printf("[pkes] Step 5 done..Scanning for APs\n", cnt);
+		wifi_manager_start_scan();
+		WM_TEST_WAIT;
+		printf("[pkes] Step 6 done..Disconnecting AP\n");
+		/* Disconnect AP */
+		wifi_manager_ap_disconnect();
+		WM_TEST_WAIT;
+		printf("[pkes] Step 7 done..Cycle finished [Round %d]\n", cnt);
+	}
+
+	res = wifi_manager_deinit();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		printf("[pkes] WiFi Manager failed to stop\n");
+		return ;
+	}
+
+	printf("Exit Wi-Fi Manager Stress Test..\n");
+
+	return ;
+}
+
+
+int
+wm_parse_commands(struct options *opt, int argc, char *argv[])
+{
+	if (argc < 3) {
+		return -1;
```

Comment:
Don't you need to support help?

Reply:
wm_process will print usage if return value is less than 0 at wm_parse_commands.